src/components/ScrollToTop/scrolltotop.tsx

```diff
@@ -0,0 +1,45 @@
+"use client";
+
+import React, { useState, useEffect } from 'react';
+
+const ScrollToTop: React.FC = () => {
+  const [isVisible, setIsVisible] = useState(false);
+
+  const toggleVisibility = () => {
+    if (window.scrollY > 300) {
+      setIsVisible(true);
+    } else {
+      setIsVisible(false);
+    }
+  };
+
+  const scrollToTop = () => {
+    window.scrollTo({
+      top: 0,
+      behavior: 'smooth',
+    });
+  };
+
+  useEffect(() => {
+    window.addEventListener('scroll', toggleVisibility);
+    return () => {
+      window.removeEventListener('scroll', toggleVisibility);
+    };
+  },  [toggleVisibility]);
+  
+  return (
+    <div className="fixed bottom-9 right-5 z-50">
+      {isVisible && (
+        <button 
+          type="button"
+          onClick={scrollToTop}
+          className="bg-pink-500 hover:bg-pink-600 text-white p-2 rounded-full h-12 w-12 flex items-center justify-center shadow-lg transition-transform transform hover:scale-110"
```

Comment:
instead of pink color, use bg-primary which auto utilizes our color scheme and on hover, scale it to 105

Reply:
thank you for letting me know!!